Px.Utils/Validation/DataValidation/DataValidatorFunctions.cs

```diff
@@ -0,0 +1,210 @@
+                };
+            }
+
+            return Array.Empty<ValidationFeedback>();
+        }
+    }
+
+    public class DataRowLengthValidator(int rowLen) : IDataValidator
+    {
+        private int _currentRowLen = 0;
+
+        /// <summary>
+        /// Validates the given token for row length.
+        /// </summary>
+        /// <param name="token">The token to validate.</param>
+        /// <returns>An enumerable collection of ValidationFeedback objects.</returns>
+        public IEnumerable<ValidationFeedback> Validate(Token token)
+        {
+        
+            switch (token.Type)
+            {
+                case TokenType.DataItemSeparator:
+                    _currentRowLen++;
+                    break;
+                case TokenType.LineSeparator:
+                    var itemNum = _currentRowLen;
+                    if (_currentRowLen != rowLen)
+                    {
+                        _currentRowLen = 0;
+                        return new[] { new ValidationFeedback(ValidationFeedbackLevel.Error, 
+                            ValidationFeedbackRule.DataValidationFeedbackInvalidRowLength, token.LineNumber, token.CharPosition,$"{rowLen},{itemNum}") };
+                    }
+                    _currentRowLen = 0;
+                    break;
+            }
+
+            return Array.Empty<ValidationFeedback>();
+        }
+    }
+
+    public class DataStringValidator : IDataValidator
+    {
+        private static readonly string[] ValidStringDataItems =
+        [
+            PxFileSyntaxConf.Default.Tokens.DataValues.DataIsMissing,
+            PxFileSyntaxConf.Default.Tokens.DataValues.DataCategoryNotApplicable,
+            PxFileSyntaxConf.Default.Tokens.DataValues.DataIsConfidential,
+            PxFileSyntaxConf.Default.Tokens.DataValues.DataIsNotAvailable,
+            PxFileSyntaxConf.Default.Tokens.DataValues.DataHasNotBeenAsked,
+            PxFileSyntaxConf.Default.Tokens.DataValues.Missing6,
+            PxFileSyntaxConf.Default.Tokens.DataValues.DataIsNone
+        ];
+
+        /// <summary>
+        /// Validates a token as a string data item.
+        /// </summary>
+        /// <param name="token">The token to be validated.</param>
+        /// <returns>An enumerable collection of <see cref="ValidationFeedback"/> objects containing any validation feedback.</returns>
+        public IEnumerable<ValidationFeedback> Validate(Token token)
+        {
+            if (token.Type != TokenType.StringDataItem) return Array.Empty<ValidationFeedback>();
+
+            if (ValidStringDataItems.Contains(token.Value))
+            {
+                return Array.Empty<ValidationFeedback>();
+            }
+
+            return new[] { new ValidationFeedback(ValidationFeedbackLevel.Error, 
+                ValidationFeedbackRule.DataValidationFeedbackInvalidString, token.LineNumber, token.CharPosition,$"{token.Value}") };
+        }
+    }
+
+    public class DataNumberDataValidator : IDataValidator
```

Comment:
NumberDataValidator?

Reply:
made it DataNumberValidator, so that all validators start with Data...